fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -182,7 +182,10 @@ def __init__(self, module: nn.Module, param_list: ParamGroups = None):
         # Init all flat_params.
         for new_p_set in self._param_sets:
             params = self._init_flatten_params(new_p_set)
-            flat_param = FlatParameter(params)
+            assert (
+                len(set(p.requires_grad for p in params)) == 1
+            ), "expects all parameters in the same parameter group of the module to have same requires_grad"
+            flat_param = FlatParameter(params, params[0].requires_grad)
```

Comment:
Nice fix! Just to confirm, without this line, what's the error msg from the unit test?

Reply:
error msg is like this, indicating gradients are not calculated correctly

E       AssertionError: For dict key 'trunk.0.weight': With rtol=0.0001 and atol=1e-05, found 1710 element(s) (out of 1728) whose difference(s) exceeded the margin of error (including 0 nan comparisons). The greatest difference was 0.00930418074131012 (-0.1777753382921219 vs. -0.16847115755081177), which occurred at index (49, 1, 2, 2).